[32mINFO    [0m integration.test_upgrade:test_upgrade.py:120 starting continuous writes to the database
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:124 checking whether writes are increasing
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:127 Get leader unit
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:131 Run pre-upgrade-check action
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:139 Inject dependency fault
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:144 Refresh the charm
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:147 Wait for upgrade to fail
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:154 Ensure continuous_writes while in failure state on remaining units
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:157 Re-run pre-upgrade-check action
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:161 Re-refresh the charm
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:164 Wait for upgrade to start
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:170 Wait for application to recover
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer/0 [idle] blocked: ready to rollback application
  pgbouncer/1 [idle] waiting: other units upgrading first...
  pgbouncer/2 [idle] waiting: other units upgrading first...
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer/0 [idle] active: 
  pgbouncer/1 [idle] active: 
  pgbouncer/2 [idle] active: 
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:174 Ensure continuous_writes after rollback procedure
[32mINFO    [0m integration.test_upgrade:test_upgrade.py:179 Checking whether no writes were lost
[32mINFO    [0m pytest_operator.plugin:plugin.py:951 Model status:

Model    Controller     Cloud/Region         Version  SLA          Timestamp
testing  concierge-lxd  localhost/localhost  3.6.10   unsupported  01:59:07Z

App                  Version  Status  Scale  Charm                Channel      Rev  Exposed  Message
pgbouncer            1.21.0   active      3  pgbouncer                           2  no       
postgresql           14.19    active      3  postgresql           14/edge      918  no       
postgresql-test-app           active      3  postgresql-test-app  latest/edge  406  no       received database credentials of the first database

Unit                    Workload  Agent  Machine  Public address  Ports     Message
postgresql-test-app/0*  active    idle   0        10.247.73.147             received database credentials of the first database
  pgbouncer/0*          active    idle            10.247.73.147             
postgresql-test-app/1   active    idle   1        10.247.73.12              received database credentials of the first database
  pgbouncer/1           active    idle            10.247.73.12              
postgresql-test-app/2   active    idle   2        10.247.73.114             received database credentials of the first database
  pgbouncer/2           active    idle            10.247.73.114             
postgresql/0            active    idle   3        10.247.73.242   5432/tcp  
postgresql/1*           active    idle   4        10.247.73.148   5432/tcp  Primary
postgresql/2            active    idle   5        10.247.73.116   5432/tcp  

Machine  State    Address        Inst id        Base          AZ             Message
0        started  10.247.73.147  juju-2ff027-0  ubuntu@22.04  runnervm3ublj  Running
1        started  10.247.73.12   juju-2ff027-1  ubuntu@22.04  runnervm3ublj  Running
2        started  10.247.73.114  juju-2ff027-2  ubuntu@22.04  runnervm3ublj  Running
3        started  10.247.73.242  juju-2ff027-3  ubuntu@22.04  runnervm3ublj  Running
4        started  10.247.73.148  juju-2ff027-4  ubuntu@22.04  runnervm3ublj  Running
5        started  10.247.73.116  juju-2ff027-5  ubuntu@22.04  runnervm3ublj  Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:957 Juju error logs:

machine-4: 01:46:18 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-4: 01:46:18 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-1: 01:49:00 ERROR unit.postgresql/1.juju-log database-peers:3: Failed to list PostgreSQL database users: could not translate host name "None" to address: Temporary failure in name resolution

machine-3: 01:49:03 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:49:03 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-5: 01:49:40 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-5: 01:49:40 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:50:05 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:50:05 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:50:40 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:50:40 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:51:15 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:51:15 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-0: 01:51:34 ERROR unit.postgresql/0.juju-log database:6: Failed to list PostgreSQL database users: connection to server at "10.247.73.242", port 5432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?

unit-pgbouncer-0: 01:58:01 ERROR unit.pgbouncer/0.juju-log charms.data_platform_libs.v0.upgrade.VersionError - {'message': 'Versions incompatible, snap 1.18.0 can not be upgraded to 15.0', 'cause': 'Upgrades only supported for specific versions, snap versions satisfying requirement ^15', 'resolution': ''}

[32mINFO    [0m pytest_operator.plugin:plugin.py:1039 Forgetting model main...